apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/interceptor/enhance/DelegateNamingResolver.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance;
+
+import net.bytebuddy.utility.RandomString;
+import org.apache.skywalking.apm.agent.core.plugin.AbstractClassEnhancePluginDefine;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.StaticMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.v2.InstanceMethodsInterceptV2Point;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.v2.StaticMethodsInterceptV2Point;
+
+import java.util.Objects;
+
+/**
+ * Generate fixed delegate field name for MethodDelegation
+ */
+public class DelegateNamingResolver {
+    private static final String PREFIX = "delegate$";
+    private static String NAME_TRAIT = "sw$";
```

Comment:
Another duplicate case, I think. This should be either from constructor or final field.

Reply:
The name trait is defined globally in `SkyWalkingAgent` , and passed into `DelegateNamingResolver `  with static method `setNameTrait( )`, how to  use final field?

Reply:
Yes, I mean that. Then why do we need a default value?

Reply:
Further, I don't think you need a static `setNameTrait` method as well. Read https://github.com/apache/skywalking-java/pull/561#discussion_r1239792018. 
Keeping class as instance level only would be helpful to understand codes and avoid bugs.